reaper.go

```diff
@@ -146,19 +156,11 @@ func reuseOrCreateReaper(ctx context.Context, sessionID string, provider ReaperP
 	reaperContainer, err := lookUpReaperContainer(context.Background(), sessionID)
 	if err == nil && reaperContainer != nil {
 		// The reaper container exists as a Docker container: re-use it
-		endpoint, err := reaperContainer.PortEndpoint(ctx, "8080", "")
+		Logger.Printf("🔥 Reaper obtained from Docker for this test session %s", reaperContainer.ID)
```

Comment:
I think this log entry should be printed just once, instead of: https://github.com/testcontainers/testcontainers-go/pull/1904/files#diff-e8ab4ec17e978db949cd15726ba0efa9c89fd714e7e6e0aa537ed7fc1bbde7fdR286

Reply:
The "original" entry was logged when lookup succeeded and we would not try to create. The added (slightly changed) one is logged if we already concluded that the reaper is not up yet but it turned out that we were "wrong". This is why the entry is logged inside of `newReaper` as we usually would not expect it from outside.
The main reason why I would add it is that we get this log entry in `newReaper` right before creation  fails:
```
2023/11/06 17:03:02 🐳 Creating container for image docker.io/testcontainers/ryuk:0.5.1
```
I thought that it would be confusing to see this message multiple times, expecting Ryuk to be started despite it _not_ starting because of recovery.
If you still want to remove the added log entry, just tell me :)

Reply:
For me the most confusing part was to see this in the logs:

> 🔥 Canceling creation - Reaper obtained from Docker for this test session 06ed5f976d4d87db94954a7505a3b7cef8e5e22afe511dabd27bfc373d99afc8

I like this PR, so I'm not blocking it at all, just trying to make the log a bit more readable. We can simply keep the PR as is updating the above log message to the original one:

> 🔥 Reaper obtained from Docker for this test session 06ed5f976d4d87db94954a7505a3b7cef8e5e22afe511dabd27bfc373d99afc8

Regarding the log you mention that appears before the container is created:

> 2023/11/06 17:03:02 🐳 Creating container for image docker.io/testcontainers/ryuk:0.5.1

it comes from the container lifecycle, in the beforeCreates hook. We could (in a follow-up) move part of this code detecting the existence of the reaper to the reaper's container lifecycle hook, trying to be smarter on what we print out. But that would be totally secondary, and not related to this PR.